src/Microsoft.Management.Deployment/PackageManager.cpp

```diff
@@ -1197,5 +1322,114 @@ namespace winrt::Microsoft::Management::Deployment::implementation
             true /*canCancelQueueItem*/, nullptr /*queueItem*/, package, options, std::move(callerProcessInfoString));
     }
 
+    winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Microsoft::Management::Deployment::AddPackageCatalogResult, double> PackageManager::AddPackageCatalogAsync(winrt::Microsoft::Management::Deployment::AddPackageCatalogOptions options)
+    {
+        LogStartupIfApplicable();
+
+        // options must be set.
+        if (!options)
+        {
+            co_return GetAddPackageCatalogResult(APPINSTALLER_CLI_ERROR_INVALID_CL_ARGUMENTS);
+        }
+
+        if (options.Name().empty() || options.SourceUri().empty())
+        {
+            co_return GetAddPackageCatalogResult(APPINSTALLER_CLI_ERROR_INVALID_CL_ARGUMENTS);
+        }
+
+        HRESULT terminationHR = S_OK;
+        try {
+
+            // Check if running as admin
+            // [NOTE:] For OutOfProc calls, the Windows Package Manager Service executes in the context initiated by the caller process,
+            //so the same admin validation check is applicable for both InProc and OutOfProc calls.
+            THROW_HR_IF(APPINSTALLER_CLI_ERROR_COMMAND_REQUIRES_ADMIN, !AppInstaller::Runtime::IsRunningAsAdmin());

```

Comment:
Is it required to check if the calling process should be running as an Admin for the packaged context? 

The scenario I want to confirm involves two processes, p1 and p2, both attempting to call OOP APIs. If p1, running as an admin, starts the WinGet COM Server initially, and p2, running in user context, tries to invoke PackageCatalogManagement APIs, 
1. will p2's requests be served by the WinGet COM Server running in p1's context, or 
2. will a new WinGet COM Server process be launched in p2's context to serve p2's requests?

Reply:
As discussed, unelevated processes should use standard COM activation and end up in an unelevated server.  Elevated processes use manual activation and end up in an elevated server.  The elevated server does not allow unelevated clients to connect.